gci/src/main/java/com/gcinterceptor/gci/HeapMonitor.java

```diff
@@ -0,0 +1,39 @@
+package com.gcinterceptor.gci;
+
+import java.lang.management.ManagementFactory;
+import java.lang.management.MemoryPoolMXBean;
+
+class HeapMonitor {
```

Comment:
This class is not only monitoring anymore, it does collections. So, let's call it Heap (remember you need to rename the file).